Source/WebKit/NetworkProcess/Downloads/cocoa/DownloadCocoa.mm

```diff
@@ -157,25 +155,24 @@
         startUpdatingProgress();
     };
 
-    sendWithAsyncReply(Messages::DownloadProxy::DidReceivePlaceholderURL(placeholderURL, span(data)), WTFMove(completionHandler));
+    sendWithAsyncReply(Messages::DownloadProxy::DidReceivePlaceholderURL(placeholderURL, span(bookmarkData), WTFMove(sandboxExtensionHandle)), WTFMove(completionHandler));
 }
 
-void Download::setFinalURL(NSURL *finalURL)
+void Download::setFinalURL(NSURL *finalURL, NSData *bookmarkData)
 {
     if (!finalURL)
         return;
 
     BOOL usingSecurityScopedURL = [finalURL startAccessingSecurityScopedResource];
 
-    NSError *error = nil;
-    NSData *data = [finalURL bookmarkDataWithOptions:0 includingResourceValuesForKeys:nil relativeToURL:nil error:&error];
-    if (error)
-        RELEASE_LOG_ERROR(Network, "Failed to create bookmark data: %{public}@", error);
+    SandboxExtension::Handle sandboxExtensionHandle;
+    if (auto handle = SandboxExtension::createHandleWithoutResolvingPath(StringView::fromLatin1(finalURL.fileSystemRepresentation), SandboxExtension::Type::ReadOnly))
+        sandboxExtensionHandle = WTFMove(*handle);
 
     if (usingSecurityScopedURL)
         [finalURL stopAccessingSecurityScopedResource];
 
-    send(Messages::DownloadProxy::DidReceiveFinalURL(finalURL, span(data)));
+    send(Messages::DownloadProxy::DidReceiveFinalURL(finalURL, span(bookmarkData), WTFMove(sandboxExtensionHandle)));
```

Comment:
I don't think we need WTFMove() for bookmakData, seems like a pointer.

Reply:
Ah, good point, will fix! I will also address the other comments in the upcoming patch.

Thanks for reviewing!